### ISSUE TITLE
Scylla does not preserve comparator type of thrift static tables

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#2573]
*Installation details*
Scylla version (or git commit hash): < 2.0

dtest, in thrift_test.py:

```python
def test_nonstandard_comparator_with_column_metadata(self):
        ks = 'Keyspace1'
        _set_keyspace(ks)
        cd = ColumnDef(_i64(4), 'LongType', None, None)
        cf = CfDef('Keyspace1', 'test_table', column_metadata=[cd], comparator_type='LongType')
        client.system_add_column_family(cf)
        print(client.describe_keyspace(ks))
```

Scylla:

```
CfDef(comment='', key_validation_class='org.apache.cassandra.db.marshal.BytesType', index_interval=None,
                  min_compaction_threshold=4, key_cache_save_period_in_seconds=None,
                  row_cache_save_period_in_seconds=None, gc_grace_seconds=864000, default_time_to_live=0,
                  memtable_flush_period_in_ms=0, default_validation_class='org.apache.cassandra.db.marshal.BytesType',
                  max_compaction_threshold=32, read_repair_chance=0.0,
                  compression_options={'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'},
                  bloom_filter_fp_chance=0.01, id=None, keyspace='Keyspace1', key_cache_size=None,
                  populate_io_cache_on_flush=None, replicate_on_write=None, subcomparator_type=None,
                  merge_shards_chance=None, row_cache_provider=None, min_index_interval=None, column_type='Standard',
                  memtable_throughput_in_mb=None, memtable_flush_after_mins=None, column_metadata=[
                    ColumnDef(index_type=None, index_name=None,
                              validation_class='org.apache.cassandra.db.marshal.LongType',
                              name='\x00\x00\x00\x00\x00\x00\x00\x04', index_options=None)], key_alias='key',
                  max_index_interval=None, dclocal_read_repair_chance=0.0, name='test_table', triggers=None,
                  speculative_retry='NONE', cells_per_row_to_cache='100', compaction_strategy_options={},
                  row_cache_keys_to_save=None, compaction_strategy='SizeTieredCompactionStrategy',
                  memtable_operations_in_millions=None, caching='all',
                  comparator_type='org.apache.cassandra.db.marshal.UTF8Type', row_cache_size=None),
```

Cassandra 2.1.17:

```
CfDef(comment='', key_validation_class='org.apache.cassandra.db.marshal.BytesType', index_interval=None,
                  min_compaction_threshold=4, key_cache_save_period_in_seconds=None,
                  row_cache_save_period_in_seconds=None, gc_grace_seconds=864000, default_time_to_live=0,
                  memtable_flush_period_in_ms=0, default_validation_class='org.apache.cassandra.db.marshal.BytesType',
                  max_compaction_threshold=32, read_repair_chance=0.0,
                  compression_options={'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'},
                  bloom_filter_fp_chance=None, id=None, keyspace='Keyspace1', key_cache_size=None,
                  populate_io_cache_on_flush=None, replicate_on_write=None, subcomparator_type=None,
                  merge_shards_chance=None, row_cache_provider=None, min_index_interval=128, column_type='Standard',
                  memtable_throughput_in_mb=None, memtable_flush_after_mins=None, column_metadata=[
                    ColumnDef(index_type=None, index_name=None,
                              validation_class='org.apache.cassandra.db.marshal.LongType',
                              name='\x00\x00\x00\x00\x00\x00\x00\x04', index_options=None)], key_alias='key',
                  max_index_interval=2048, dclocal_read_repair_chance=0.0, name='test_table', triggers=[],
                  speculative_retry='NONE', cells_per_row_to_cache='0', compaction_strategy_options={},
                  row_cache_keys_to_save=None,
                  compaction_strategy='org.apache.cassandra.db.compaction.SizeTieredCompactionStrategy',
                  memtable_operations_in_millions=None, caching='KEYS_ONLY',
                  comparator_type='org.apache.cassandra.db.marshal.LongType', row_cache_size=None),
```

The problem is that `create_table_from_mutations()` doesn't set `_regular_column_name_type` based on the `"comparator"` column. It defaults to utf8_type.

**Comment by duarten** [comment]
@tgrabiec How urgent is this?

**Comment by tgrabiec** [comment]
Noone complained so far, so I guess not very.

**Comment by tgrabiec** [comment]
If you try to create the following table:

```python
cf = CfDef(ks_name, 'test_table',
                   column_metadata=[
                       ColumnDef(_i64(4), 'Int32Type'),
                       ColumnDef(_i64(-2), 'Int32Type')
                   ],
                   key_validation_class='UTF8Type',
                   comparator_type='LongType',
                   default_validation_class='UTF8Type')
```

insertion will fail right away because -2 fails the utf8 validation. 

You also won't be able to connect with `cqlsh` any more due to the following driver error, until you some how remove that table from the schema tables:

```
cassandra.cluster.NoHostAvailable: ('Unable to connect to any servers', {'127.0.0.1': UnicodeDecodeError('utf8', '\xff\xff\xff\xff\xff\xff\xff\xfe', 0, 1, 'invalid start byte')})
```

**Comment by duarten** [comment]
On master this fails in a different way, because we insert the synthetic compact_value_name column, whose comparator is different than that of the specified one.

**Comment by tgrabiec** [comment]
Yes, that's https://github.com/scylladb/scylla/issues/2584 on which I'm working.

**Comment by duarten** [comment]
I solved it by making regular_column_name_type compatible with cassandra 3.0, that is, apply it only to static columns in static compact tables. In cassandra 3.0, the synthetic compact value name column also uses the default comparator and validator.

**Comment by tgrabiec** [comment]
@duarten This issue is about fixing 1.7

**Comment by duarten** [comment]
@tgrabiec Oh, gotcha. Is it worth it though? We'd have to insert some synthetic column to preserve the comparator type, I guess. I don't see a trivial solution.

**Comment by tgrabiec** [comment]
I don't think this is urgent to fix.

We shouldn't have to add any extra columns. The comparator can be derived from the schema object at any time using cell_comparator::to_sstring(). The problem is that the code which reads schema from schema tables doesn't set _regular_column_name_type based on the persisted comparator type name.

**Comment by duarten** [comment]
Ah, I missed that we persisted the comparator type. Should be easy then.
I'll fix it since it allows us to better test migration.

**Comment by duarten** [comment]
Fixed by 6501bf8.
